osutil/flock_test.go

```diff
@@ -127,9 +128,11 @@ func (s *flockSuite) TestLockUnlockNonblockingWorks(c *C) {
 
 	// Use the "flock" command to grab a lock for 9999 seconds in another process.
 	lockPath := filepath.Join(c.MkDir(), "lock")
-	cmd := exec.Command("flock", "--exclusive", lockPath, "sleep", "9999")
+	sleeperKillerPath := filepath.Join(c.MkDir(), "pid")
+	// we can't use --no-fork because we still support 14.04
+	cmd := exec.Command("flock", "--exclusive", lockPath, "-c", fmt.Sprintf("echo \"kill $$\" > %s && exec sleep 30", sleeperKillerPath))
```

Comment:
Looks like we could put the command in a new process group and then do `kill(-pgid, KILL)` ?

This seems to work locally for me:
```
@@ -128,11 +128,20 @@ func (s *flockSuite) TestLockUnlockNonblockingWorks(c *C) {
 
        // Use the "flock" command to grab a lock for 9999 seconds in another process.
        lockPath := filepath.Join(c.MkDir(), "lock")
-       sleeperKillerPath := filepath.Join(c.MkDir(), "pid")
        // we can't use --no-fork because we still support 14.04
-       cmd := exec.Command("flock", "--exclusive", lockPath, "-c", fmt.Sprintf("echo \"kill $$\" > %s && exec sleep 30", sleeperKillerPath))
+       cmd := exec.Command("flock", "--exclusive", lockPath, "-c", "exec sleep 30")
+       cmd.SysProcAttr = &syscall.SysProcAttr{
+               Setpgid: true,
+               Pgid:    0,
+       }
        c.Assert(cmd.Start(), IsNil)
-       defer func() { exec.Command("/bin/sh", sleeperKillerPath).Run() }()
+       defer func() {
+               pgid, err := syscall.Getpgid(cmd.Process.Pid)
+               if err != nil && err != syscall.ENOENT {
+                       c.Fatalf("unexpected error during process cleanup: %v", err)
+               }
+               syscall.Kill(-pgid, syscall.SIGKILL)
+       }()
 
        // Give flock some chance to create the lock file.
        for i := 0; i < 10; i++ {
```

Reply:
Added some tweaks to use process group here and in kdialog: https://github.com/snapcore/snapd/compare/master...bboozzoo:bboozzoo/process-fun-fun 
Perhaps we should use this pattern more widely.

Reply:
@bboozzoo we should merge what mvo made and iterate with your approach.